ssv/runner.go

```diff
@@ -135,56 +139,65 @@ func (b *BaseRunner) baseConsensusMsgProcessing(runner Runner, msg *qbft.SignedM
 		return false, nil, nil
 	}
 
-	if decideCorrectly, err := b.didDecideCorrectly(prevDecided, decidedMsg); !decideCorrectly {
+	if decideCorrectly, err := b.didDecideCorrectly(prevDecided, decidedSignedMsg); !decideCorrectly {
 		return false, nil, err
 	}
 
 	// decode consensus data
-	decidedValue = &types.ConsensusData{}
-	if err := decidedValue.Decode(decidedMsg.FullData); err != nil {
+	switch runner.(type) {
```

Comment:
maybe should be in the runner itself?

Reply:
maybe y using the `encoding` interface this can be manages

Reply:
This can be a type like RunnerType with 2 options for now 
`Committee`
`Validaror`

so `Validator` will be default.

since only two options can be also managed with a boolean.

I agree with alon here that the switching on the type is not readable and too language specific to be part of the spec.

Reply:
done Alon's way